src/source/TsGenerator.scala

```diff
@@ -72,7 +72,7 @@ class TsGenerator(spec: Spec) extends Generator(spec) {
   }
 
   def toTsType(tm: MExpr, addNullability: Boolean = true): String = {
-    def args(tm: MExpr) = if (tm.args.isEmpty) "" else tm.args.map(f).mkString("<", ", ", ">")
+    def args(tm: MExpr) = if (tm.args.isEmpty) "" else tm.args.map(arg => toTsType(arg, addNullability)).mkString("<", ", ", ">")
```

Comment:
This is the fix. The only difference between `f` and `toTsType` is it takes into account the optional logic
```
(if (addNullability) nullityAnnotation(tm) else "")
```

We capture `addNullability` from the outer scope, as it should be the same for all the args.